modin/pandas/groupby.py

```diff
@@ -116,24 +116,33 @@ def __getattr__(self, key):
     def ngroups(self):
         return len(self)
 
-    def skew(self, **kwargs):
-        return self._apply_agg_function(lambda df: df.skew(**kwargs))
+    def skew(self, *args, **kwargs):
+        return self._wrap_aggregation(
+            type(self._query_compiler).groupby_skew,
+            agg_args=args,
+            agg_kwargs=kwargs,
+            numeric_only=True,
+        )
 
     def ffill(self, limit=None):
         return self._default_to_pandas(lambda df: df.ffill(limit=limit))
 
     def sem(self, ddof=1):
         return self._default_to_pandas(lambda df: df.sem(ddof=ddof))
 
-    def mean(self, *args, **kwargs):
-        return self._apply_agg_function_check_index(lambda df: df.mean(*args, **kwargs))
+    def mean(self, numeric_only=None):
+        return self._check_index(
```

Comment:
Why is `_check_index` sometimes used and sometimes not?

Reply:
`_check_index` and `_check_index_name` do special casing for `as_index=False` when grouping on an index level for certain operations (mimic pandas behavior). These functions were introduced at #3579, the reason why some functions need special handling and others don't is explained in [this comment](https://github.com/modin-project/modin/pull/3579#discussion_r739127976).

At this PR I  just replaced `_apply_agg_func_check_*` with the `_check_*` everywhere in this file (after your comment I spot one place where I missed the `_check_index_name` call, fixed it at the last commit).